pinns/src/sysctl.c

```diff
@@ -0,0 +1,105 @@
+#define _GNU_SOURCE
+
+#include <string.h>
+#include <fcntl.h>
+
+#include "sysctl.h"
+#include "utils.h"
+
+static int separate_sysctl_key_value (char* sysctl_key_value, char** sysctl_key, char** sysctl_value);
+static int write_sysctl_to_file (char * sysctl_key, char* sysctl_value);
+const char *sysctl_delim = "+";
+
+int configure_sysctls (char * const sysctls) {
+  char* sysctl = strtok(sysctls, sysctl_delim);
+  char* key = NULL;
+  char* value = NULL;
+  while (sysctl) {
+    if (separate_sysctl_key_value (sysctl, &key, &value) < 0)
+      return -1;
+
+    if (write_sysctl_to_file (key, value) < 0)
+      return -1;
+	
+	sysctl = strtok (NULL, sysctl_delim);
+  }
+
+  return 0;
+}
+
+// sysctl_key_value should be in the form `'key=value'`
+static int separate_sysctl_key_value (char* sysctl_key_value, char** sysctl_key, char** sysctl_value)
+{
+  // begin by stripping the `'`, we now have `key=value'`
+  if (*sysctl_key_value == '\'') {
+    sysctl_key_value++;
+  }
+
+  // now find the `=` and convert it to a delimiter
+  char * equals_token = strchr (sysctl_key_value, '=');
+  if (!equals_token) {
+    nwarnf ("sysctl must be in the form of 'key=value'; '=' missing from %s", sysctl_key_value);
+    return -1;
+  }
+  // we now have `key\0value'`
+  *equals_token = '\0';
+
+  // sysctl_key is now key
+  *sysctl_key = sysctl_key_value;
+
+  // equals_token is now value'
+  ++equals_token;
+
+  // if sysctl ends in a ', we should strip it
+  char* ending_char = strchr (equals_token, '\'');
```

Comment:
1. This should be stripped only if we found the opening `'` earlier.
2. What you say ("if sysctl ends in a ', we should strip it") is not what you do here (find `'` and strip it with the remainder.
3. I think shell will strip those single quotes for you.

Reply:
I am going to leave the `'` handling (3), but everything else was addressed